pkg/dependencies/factory.go

```diff
@@ -284,7 +289,7 @@ func (f *Factory) WithExecutableBuilder() *Factory {
 		if f.executablesConfig.useDockerContainer {
 			image := f.executablesConfig.image
 			if f.registryMirror != nil {
-				image = urls.ReplaceHost(f.executablesConfig.image, f.registryMirror.endpoint)
+				image = urls.ReplaceHost(f.executablesConfig.image, f.registryMirror.endpoints[constants.DefaultRegistry])
```

Comment:
I think we should check the mappings and only do the replace if there is one than matches the image.

I think we could benefit from a service object that does this for us that we can inject in other places, instead of keep using `ReplaceHost` everywhere.